corehq/apps/translations/app_translations/upload_module.py

```diff
@@ -0,0 +1,291 @@
+        # It's a case list registration form label. Don't add it to condensed rows
+        elif row['case_property'] == 'case_list_form_label':
+            case_list_form_label = row
+
+        # If it's a tab header, don't add it to condensed rows
+        elif re.search(r'^Tab \d+$', row['case_property']):
+            index = int(row['case_property'].split(' ')[-1])
+            if index < len(detail_tab_headers):
+                detail_tab_headers[index] = row
+            else:
+                msgs.append((
+                    messages.error,
+                    _("Expected {0} case detail tabs in sheet {1} but found row for Tab {2}. "
+                      "No changes were made for sheet {1}.").format(
+                          len(detail_tab_headers),
+                          sheet.worksheet.title,
+                          index
+                      )
+                ))
+
+        # It's a normal case property
+        else:
+            row['id'] = row['case_property']
+            condensed_rows.append(row)
+
+    return (condensed_rows, detail_tab_headers, case_list_form_label)
+
+
+def _remove_description_from_case_property(row):
+    return re.match('.*(?= \()', row['case_property']).group()
+
+
+def _check_for_detail_length_errors(module, condensed_rows):
+    errors = []
+
+    list_rows = [row for row in condensed_rows if row['list_or_detail'] == 'list']
+    detail_rows = [row for row in condensed_rows if row['list_or_detail'] == 'detail']
+    short_details = list(module.case_details.short.get_columns())
+    long_details = list(module.case_details.long.get_columns())
+    partial_upload = False
+
+    for expected_list, received_list, word in [
+        (short_details, list_rows, "list"),
+        (long_details, detail_rows, "detail")
+    ]:
+        if len(expected_list) != len(received_list):
+            # if a field is not referenced twice in a case list or detail, then
+            # we can perform a partial upload using field (case property) as a key
+            number_fields = len({detail.field for detail in expected_list})
+            if number_fields == len(expected_list) and toggles.ICDS.enabled(app.domain):
+                partial_upload = True
+                continue
+            errors.append(_("Expected {0} case {3} properties in sheet {2}, found {1}. "
+                "No case list or detail properties for sheet {2} were updated").format(
+                    len(expected_list),
+                    len(received_list),
+                    sheet.worksheet.title,
+                    word
+                )
+            )
+
+    return (errors, partial_upload)
+
+
+def _get_module_from_sheet_name(app, sheet):
+    module_index = int(sheet.worksheet.title.replace("module", "")) - 1
+    return app.get_module(module_index)
+
+
+def _has_at_least_one_translation(row, prefix, langs):
+    """
+    Returns true if the given row has at least one translation.
+
+    >> has_at_least_one_translation(
```

Comment:
I know you didn't write this docstring, but is it intentionally not a doctest? 

If you change `>>` to `>>>`, `true` to `True` and `false` to `False`, then you can get the test suite to [check the docstring](https://github.com/dimagi/commcare-hq/blob/868436c0a4c1a4daaeefd6c89ef6310af3d542e7/corehq/sql_db/tests/test_backports.py#L50) is correct with 

```python
def test_doctests():
    results = doctest.testmod(upload_module)
    assert results.failed == 0
```

Reply:
902c98d4013